mlog/testing.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package mlog
+
+import (
+	"testing"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// testingWriter is an io.Writer that writes through t.Log
+type testingWriter struct {
+	tb testing.TB
+}
+
+func (tw *testingWriter) Write(b []byte) (int, error) {
```

Comment:
Testing this branch, it looks like `t.Log` prepends `testing.go: 20` in an attempt to identify the "source" of the log. This makes sense, as it's probably expecting `t.Log` to come from a test file whereas we're obviously latching onto that in a unique way. 

Adding `t.Helper` remove this frame from the stack, but then it just pops back to something deeper in `zap`. I don't think there's a solution to this, but just noting for completeness.

Reply:
I'd like to look into this and see if there is a fix, I thought zap would pass up the caller too when it calls the write function, but now that you point it out, i think that's not how it works.